parts/k8s/kubernetesmasterresources.t

```diff
@@ -894,7 +904,7 @@
           {{end}}
         },
         "storageProfile": {
-          {{if not UseMasterCustomImage}}
+          {{if and (not UseMasterCustomImage) IsKubernetes}}
```

Comment:
@kargakis can we prevent specification of both an imageref and a plan in validation instead?

Reply:
Seems irrelevant to this change? Background: this is adding a disk for etcd that we don't configure in openshift-ansible so I am disabling it for us altogether.

Reply:
Wonder whether we should discuss this in a separate issue or card? Wonder whether we should be making use of that etcd disk?

Reply:
+1 worth at least discussing at the next arch call

Reply:
Let's change `IsKubernetes` to `not IsOpenshift`. I assume that's the intention here. It might be confusing to have `IsKubernetes` in a kubernetes template file.

Reply:
@CecileRobertMichon I agree, any chance this can be done as a follow-up?  @kargakis is blocked on this for his CI work and would lose most of another day if we wait.  Also I opened #2750 to handle all instances of this issue

Reply:
@jim-minter @CecileRobertMichon that's fine, thanks for tracking this work generally. Let's aim to get #2750 in place this week so it makes it into v0.17.0 (to avoid unnecessary incremental changes to `kubernetesmasterresources.t` across minor versions

Reply:
Opened https://github.com/Azure/acs-engine/pull/2765